### DISCUSSION TITLE
Handsontable cross sheets reference

### DISCUSSION THREAD

**Discussion by jacksontong** [handsontable/hyperformula#1065] [General]
We have an SPA app where each page has its own hot sheet. There will be a summary sheet that will calculate data using value from other sheets. Is it possible to use Hyperformula on the summary sheet and request the cross sheet reference on the server so that we don't have to download all the sheets data to the summary sheet page?

https://handsontable.com/docs/formula-calculation/#basic-multi-sheet-example

```js
// we want the data to stay on the server
const data1 = [
  ['10.26', null, 'Sum', '=SUM(A:A)'],
  ['20.12', null, 'Average', '=AVERAGE(A:A)'],
  ['30.01', null, 'Median', '=MEDIAN(A:A)'],
  ['40.29', null, 'MAX', '=MAX(A:A)'],
  ['50.18', null, 'MIN', '=MIN(A1:A5)'],
];

// summary
const data2 = [
  ['Is A1 in Sheet1 > 10?', '=IF(Sheet1!A1>10,"TRUE","FALSE")'],
  ['Is A:A in Sheet > 150?', '=IF(SUM(Sheet1!A:A)>150,"TRUE","FALSE")'],
  ['How many blank cells are in the Sheet1?', '=COUNTBLANK(Sheet1!A1:D5)'],
  ['Generate a random number', '=RAND()'],
  ['Number of sheets in this workbook', '=SHEETS()'],
];
```

**Comment by warpech** [comment]
HyperFormula does not have built-in "resumability" (way to split the data graph and calculate a part of it in one instance, then resume to calculate another part in another instance)

To state the obvious, HyperFormula needs to have all of the data to calculate the results. I think that the reason for this is clear - to evaluate an expression like `=IF(SUM(Sheet1!A:A)>150,"TRUE","FALSE")`, you really need all of the column A in Sheet1.

I think that your problem can be mitigated in the app architecture. Have you fought about the following:

1. Run HyperFormula on the server that has access to all of the page pages, and only send the results, not the expressions, to the client.
2. Or, use named expressions as an intermediate "cache" of values between the pages: https://hyperformula.handsontable.com/guide/named-expressions.html. You could export expression values from page 1 and import them to page 2 as named expressions, not as formulas. This will only work if your expressions don't change over time.
